### PR TITLE
fix(TDP-11502): filters with special chars -> escape single quote

### PR DESCRIPTION
**Pull request by abenali-talend**
**What is the problem this Pull Request is trying to solve?**
 TDP-11502 : filters with special chars
**What is the chosen solution to this problem?**
  escape single quote character in TQL to DSEL converter"
**Link to the JIRA issue**
https://jira.talendforge.org/browse/TDP-11502
 
**Please check if the Pull Request fulfills these requirements**
- [ ] The PR commit message follows our [guidelines](https://github.com/Talend/daikon/blob/master/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features, coverage should be over 75% in the new code)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in JIRA), if any, are all linked or available in the Pull Request

 
**[ ] This Pull Request introduces a breaking change**

### CODE REVIEW

File: daikon-tql/daikon-tql-dsel/src/main/java/org/talend/tqldsel/tqltodsel/AbstractTqlToDselVisitor.java [hunk 1]
```diff
@@ -32,6 +27,11 @@
 import org.talend.tql.model.TqlElement;
 import org.talend.tql.visitor.IASTVisitor;
 
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
```
[the remark this answers is not in the document]
Comment (eguillossou) [reply daikon-tql/daikon-tql-dsel/src/main/java/org/talend/tqldsel/tqltodsel/AbstractTqlToDselVisitor.java:34]: Related to automatic formatting that will reorder imports so not really linked to current PR.

File: daikon-tql/daikon-tql-dsel/src/main/java/org/talend/tqldsel/tqltodsel/AbstractTqlToDselVisitor.java [hunk 2]
```diff
@@ -90,7 +89,7 @@ public ELNode visit(LiteralValue elt) {
         case DECIMAL:
             return new ELNode(ELNodeType.DECIMAL_LITERAL, elt.getValue());
         case QUOTED_VALUE:
-            return new ELNode(ELNodeType.STRING_LITERAL, elt.toQueryString());
+            return new ELNode(ELNodeType.STRING_LITERAL, "'" + escapeSingleQuotes(elt.getValue()) + "'");
```
Comment (eguillossou) [inline daikon-tql/daikon-tql-dsel/src/main/java/org/talend/tqldsel/tqltodsel/AbstractTqlToDselVisitor.java:92]: Maybe, " ' " can be included in method?

**Review by eguillossou — COMMENTED** [review]
read

File: daikon-tql/daikon-tql-dsel/src/main/java/org/talend/tqldsel/tqltodsel/AbstractTqlToDselVisitor.java (hunk 2 above)
Comment (sgandon) [reply daikon-tql/daikon-tql-dsel/src/main/java/org/talend/tqldsel/tqltodsel/AbstractTqlToDselVisitor.java:92]: good idea

**Review by eguillossou — DISMISSED** [review]

**Review by eguillossou — APPROVED** [review]

**Pull request merged**